src/settings/LoanHistory/LoanHistoryForm.js

```diff
@@ -0,0 +1,178 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  injectIntl,
+  intlShape,
+  FormattedMessage,
+} from 'react-intl';
+import {
+  Field,
+  FieldArray,
+} from 'redux-form';
+
+import stripesForm from '@folio/stripes/form';
+import {
+  Button,
+  Checkbox,
+  Col,
+  Pane,
+  Row,
+  Headline,
+  RadioButton,
+} from '@folio/stripes/components';
+
+import { Period } from '../components';
+import optionsGenerator from '../utils/options-genarator';
+import {
+  closingTypes,
+  closingTypesMap,
+  intervalPeriods,
+} from '../../constants';
+
+import css from './LoanHistoryForm.css';
+
+class LoanHistoryForm extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    pristine: PropTypes.bool,
+    submitting: PropTypes.bool,
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+    initialValues: PropTypes.oneOfType([PropTypes.array]),
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.generateOptions = optionsGenerator.bind(null, props.intl.formatMessage);
+    this.selectedPeriods = intervalPeriods.filter(period => props.initialValues.selectedPeriodsValues.includes(period.value));
+  }
+
+  onSave = data => this.props.onSubmit({ loan_history: JSON.stringify(data) });
+
+  getLastMenu = () => {
+    const {
+      pristine,
+      submitting,
+    } = this.props;
+
+    return (
+      <Button
+        data-test-loan-history-save-button
+        type="submit"
+        disabled={pristine || submitting}
+        marginBottom0
+      >
+        <FormattedMessage id="ui-circulation.save" />
```

Comment:
Please use `stripes-core.button.save`. There are many common/shared translation keys in stripes-core so if the label you need sounds generic, take a look there first.